packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -359,6 +385,51 @@ export class NodeProxyTarget extends ProxyTarget<Anchor> {
             .map((fieldKey) => this.proxifyField(fieldKey, false))
             .values();
     }
+
+    public createField(fieldKey: FieldKey, newContent: ITreeCursor): EditableField | undefined {
+        assert(!this.has(fieldKey), "The field already exists.");
+        const fieldKind = this.lookupFieldKind(fieldKey);
+        const path = this.cursor.getPath();
+        switch (fieldKind.multiplicity) {
+            case Multiplicity.Optional: {
+                if (this.context.setOptionalField(path, fieldKey, newContent, true))
+                    return this.proxifyField(fieldKey, false);
+            }
+            case Multiplicity.Sequence: {
+                if (this.context.insertNodes(path, fieldKey, 0, newContent))
+                    return this.proxifyField(fieldKey, false);
+            }
+            case Multiplicity.Value:
+                fail("It is invalid to create fields of kind `value` as they should always exist.");
+            default:
+                fail("`Forbidden` fields may not be created.");
+        }
+    }
+
+    public deleteField(fieldKey: FieldKey): void {
+        const fieldKind = this.lookupFieldKind(fieldKey);
+        const path = this.cursor.getPath();
+        this.cursor.enterField(fieldKey);
+        const length = this.cursor.getFieldLength();
+        this.cursor.exitField();
+        switch (fieldKind.multiplicity) {
+            case Multiplicity.Optional: {
+                // TODO: `set` an existing optional field to `undefined` is not working
```

Comment:
I will look into this to confirm there's no workaround we can use. Just leaving this as a note for myself.

Reply:
This is being addressed as part of #12819